observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/client/odm/OdmRegistryClientImpl.java

```diff
@@ -1,61 +1,59 @@
-    private final String odmPlatformUrl;
+    private String baseUrl;
+    private final RestUtils restUtils;
     private final RestTemplate restTemplate;
-    private final ObjectMapper objectMapper;
 
-    public OdmRegistryClientImpl(RestTemplate restTemplate, ObjectMapper objectMapper, String registryUrl) {
+    public OdmRegistryClientImpl(RestTemplate restTemplate, String baseUrl) {
+        this.restUtils = new RestUtils(restTemplate);
+        this.baseUrl = baseUrl;
         this.restTemplate = restTemplate;
-        this.objectMapper = objectMapper;
-        this.odmPlatformUrl = registryUrl;
     }
 
-    //Get array schemas ID
-    public List<Integer> getSchemasId(String portID) {
+    @Override
+    public List<ExternalComponentResource> getApi(String apiName, String apiVersion) {
         try {
-            return restTemplate.exchange(
-                    String.format("%s/api/v1/pp/registry/apis/%s/endpoints", odmPlatformUrl, portID),
+            String url = UriComponentsBuilder.fromHttpUrl(String.format("%s/api/v1/pp/registry/apis", baseUrl))
+                    .queryParam("name", apiName)
+                    .queryParam("version", apiVersion)
+                    .build()
+                    .toUriString();
+            //This probably will change when refactored from List to Page
+            ResponseEntity<String> responseEntity = restTemplate.exchange(
+                    url,
                     HttpMethod.GET,
                     HttpEntity.EMPTY,
-                    new ParameterizedTypeReference<List<Integer>>() {
-                    }).getBody();
-        } catch (HttpClientErrorException e) {
-            throw new RuntimeException(e); // TODO: right errors
-            //throw new MetaServiceException("Unable to get array of schemas ID: " + e.getResponseBodyAsString());
-        }
-    }
+                    String.class
+            );
+            //Esoteric method to handle bad api signature
+            return new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                    .readValue(responseEntity.getBody(), new TypeReference<>() {
+                    });
 
-    //Get Schema from given ID
-    public String getSchemaContent(Integer schemaID) {
-        try {
-            String responseEntity = restTemplate.exchange(
-                    String.format("%s/api/v1/pp/registry/schemas/%d", odmPlatformUrl, schemaID),
-                    HttpMethod.GET,
-                    HttpEntity.EMPTY,
-                    String.class).getBody();
-            return objectMapper.readValue(responseEntity, SchemaRes.class).getContent();
-        } catch (HttpClientErrorException e) {
-            throw new RuntimeException(e); // TODO: right errors
-            //throw new MetaServiceException("Unable to get array of schemas ID: " + e.getResponseBodyAsString());
+        } catch (ClientException | ClientResourceMappingException e) {
```

Comment:
Valuta se inserire codici di errori specifici (come fatto per il progetto Policy Engine OPA), o utilizza ObserverApiStandardErrors del modulo observer-api di odm-platform in modo che sia chiaro che l'errore proviene da un observer